[rush] `rush-pnpm outdated` does not work

## Summary

`rush-pnpm outdated` does not work (prints nothing).

## Repro steps

Run `rush-pnpm outdated`.

  **Expected result:** Outdated dependencies

  **Actual result:** Crickets

## Details

As noted elsewhere, `pnpm outdated -r -C common/temp` does work, but this requires us to have `pnpm` installed globally. Ideally, we should be using `rush` for everything, and Rush should be managing pnpm for us and shielding us from that.

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | 5.131.5 |
| `rushVersion` from rush.json? | 5.131.5 |
| `useWorkspaces` from rush.json? | N/A |
| Operating system? | Any |
| Would you consider contributing a PR? | No |
| Node.js version (`node -v`)? | 20.11.1 |

I'm getting this error:

```
 ERR_PNPM_NO_IMPORTER_MANIFEST_FOUND  No package.json (or package.yaml, or package.json5) was found in "<repo root path>".
```

Looks like it may not be running in the right folder. It probably expects to run with a cwd of `<repo root>/common/temp`. Probably an easy fix.

--- a/common/changes/@microsoft/rush/fix-rush-pnpm-recursive-queries_2026-06-07-06-00.json
+++ b/common/changes/@microsoft/rush/fix-rush-pnpm-recursive-queries_2026-06-07-06-00.json
@@ -1,0 +1,11 @@
+{
+  "changes": [
+    {
+      "comment": "Default `rush-pnpm outdated` and `rush-pnpm why` to recursive workspace queries.",
+      "type": "patch",
+      "packageName": "@microsoft/rush"
+    }
+  ],
+  "packageName": "@microsoft/rush",
+  "email": "262980978+kiranmagic7@users.noreply.github.com"
+}

--- a/common/changes/@microsoft/rush/fix-rush-pnpm-recursive-queries_2026-06-07-06-00.json
+++ b/common/changes/@microsoft/rush/fix-rush-pnpm-recursive-queries_2026-06-07-06-00.json
@@ -2,7 +2,7 @@
   "changes": [
     {
       "comment": "Default `rush-pnpm outdated` and `rush-pnpm why` to recursive workspace queries.",
-      "type": "patch",
+      "type": "minor",
       "packageName": "@microsoft/rush"
     }
   ],

--- a/libraries/rush-lib/src/cli/RushPnpmCommandLineParser.ts
+++ b/libraries/rush-lib/src/cli/RushPnpmCommandLineParser.ts
@@ -36,6 +36,25 @@ import { EnvironmentVariableNames } from '../api/EnvironmentConfiguration';
 import { initializeDotEnv } from '../logic/dotenv';
 
 const RUSH_SKIP_CHECKS_PARAMETER: string = '--rush-skip-checks';
+const RUSH_PNPM_RECURSIVE_DEFAULT_COMMANDS: Set<string> = new Set(['outdated', 'why']);
+
+function _hasRecursiveFlag(pnpmArgs: string[]): boolean {
+  return pnpmArgs.some((arg) => arg === '-r' || arg === '--recursive' || arg.startsWith('--recursive='));
+}
+
+function _hasGlobalFlag(pnpmArgs: string[]): boolean {
+  return pnpmArgs.some((arg) => arg === '-g' || arg === '--global' || arg.startsWith('--global='));
+}
+
+function _addDefaultRecursiveFlagIfNeeded(commandName: string, pnpmArgs: string[]): void {
+  if (
+    RUSH_PNPM_RECURSIVE_DEFAULT_COMMANDS.has(commandName) &&
+    !_hasRecursiveFlag(pnpmArgs) &&
+    !_hasGlobalFlag(pnpmArgs)
+  ) {
+    pnpmArgs.splice(1, 0, '--recursive');
+  }
+}
 
 /**
  * Options for RushPnpmCommandLineParser
@@ -253,6 +272,7 @@ export class RushPnpmCommandLineParser {
       }
 
       this._commandName = commandName;
+      _addDefaultRecursiveFlagIfNeeded(commandName, pnpmArgs);
 
       // Warn about commands known not to work
       /* eslint-disable no-fallthrough */

--- a/libraries/rush-lib/src/cli/test/RushPnpmCommandLineParser.test.ts
+++ b/libraries/rush-lib/src/cli/test/RushPnpmCommandLineParser.test.ts
@@ -1,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { RushPnpmCommandLineParser } from '../RushPnpmCommandLineParser';
+
+interface IRushPnpmCommandLineParserInternals {
+  _validatePnpmUsageAsync(pnpmArgs: string[]): Promise<void>;
+}
+
+async function validatePnpmArgsAsync(pnpmArgs: string[]): Promise<string[]> {
+  const parser: IRushPnpmCommandLineParserInternals = Object.create(RushPnpmCommandLineParser.prototype);
+  await parser._validatePnpmUsageAsync(pnpmArgs);
+  return pnpmArgs;
+}
+
+describe(RushPnpmCommandLineParser.name, () => {
+  it('adds recursive mode to workspace query commands by default', async () => {
+    await expect(validatePnpmArgsAsync(['outdated'])).resolves.toEqual(['outdated', '--recursive']);
+    await expect(validatePnpmArgsAsync(['why', '@rushstack/node-core-library'])).resolves.toEqual([
+      'why',
+      '--recursive',
+      '@rushstack/node-core-library'
+    ]);
+  });
+
+  it('does not duplicate explicit recursive flags', async () => {
+    await expect(validatePnpmArgsAsync(['outdated', '-r'])).resolves.toEqual(['outdated', '-r']);
+    await expect(
+      validatePnpmArgsAsync(['why', '--recursive', '@rushstack/node-core-library'])
+    ).resolves.toEqual(['why', '--recursive', '@rushstack/node-core-library']);
+  });
+
+  it('does not force recursive mode for global outdated checks', async () => {
+    await expect(validatePnpmArgsAsync(['outdated', '--global'])).resolves.toEqual(['outdated', '--global']);
+  });
+});

